### PR TITLE
[SofaFramework] Create Sofa.Config module

### LINKED ISSUE #1667: Create a Sofa.Config package
This will contain all the macros/configuration headers needed by all Sofa (config.h, etc)
For now, those things are contained are in:
 - SofaFramework
 - SofaHelper
 - Sofa (at the root)
 - maybe somewhere else ?
 
It will solve the issue raised at #1661

### PR DESCRIPTION
**Pull request by fredroy**
Fix #1667 , #1661 and much more...

# TL;DR
Sofa.Config is a new module containing all the configuration necessary for compiling the SofaFramework and the subsequent modules.
It allows to spread efficiently all the flags/config/whatever to the cmake modules using it.

# Description
A lot of CMake configuration/flags are spread everywhere in the code base: root CMakefile, Soframework CMakefile, SofaHelper CMakefile.
Moreover, a lot of compiler options are not set by **target** (CMake notion), so those are to added EVERYTIME for each projet/plugin (see SofaPython3): c++17, MSVC special flags, etc.

# Solution
This PR regroups all CMake config code and SofaFramework is no more the "bootstrapper" or whatever it was not.
If you linking against anything from SOFA (most likely at least SofaHelper), the configuration will spread into your project automatically.
This module will be the lowest in the depencency chain (was a bastard mix of SofaFramework/SofaHelper before)

# WIP
* ~~Some CMake file to clean~~

* ~~Add no-attribute flags from #1682~~

* Some variables are still not linked to Sofa.Config like Sofa_VERSION: it was in the root CMakefile before, so all the projects know this variable.
For now, Sofa.Config set those global variables **PARENT_SCOPE** but this is ugly in my opinion, and not "Modern CMake"
One solution is to link this variable to a property of Sofa.Config, and retrieve this property when needed. But it seems a bit bothersome.
```
get_target_property(Sofa_VERSION ${TARGET} Sofa_VERSION_PROPERTY)
```
Any opinion ?

* I guess there will other things to do later like removing all the tests related code/config into an eventual Sofa.Test, etc
______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

**Comment by damienmarchal** [comment]
Hi @fredroy 

The proposed changes sounds meaningfull to me, I didn't looked to much in the CMakeLists.txt details so I let Guillaume gives his opinion.

**Comment by fredroy** [comment]
[ci-build][force-full-build]

**Comment by fredroy** [comment]
[ci-build][force-full-build]

**Comment by damienmarchal** [comment]
So what about this PR ? Should we do something ?
